test/cadet/accounts/accounts_test.exs

```diff
@@ -21,15 +21,29 @@ defmodule Cadet.AccountsTest do
   use Cadet.DataCase
   use ExVCR.Mock, adapter: ExVCR.Adapter.Hackney
 
-  alias Cadet.{Accounts, Repo}
-  alias Cadet.Accounts.Query
+  alias Cadet.{Accounts, Accounts.Query, Repo}
 
   @token if System.get_env("TOKEN"), do: System.get_env("TOKEN"), else: "token"
 
   setup_all do
     HTTPoison.start()
   end
 
+  describe "get or create user" do
+    test "get or create exisiting user" do
+      user = insert(:user, %{nusnet_id: "E0123456"})
+      {:ok, result} = Accounts.get_or_create_user(user.name, user.role, user.nusnet_id)
+
+      assert result == user
+    end
+
+    test "get or create new user" do
+      {:ok, user} = Accounts.get_or_create_user("happy user", :staff, "E0123456")
```

Comment:
You should do something similar as the previous test except using `build` instead of `create`